src/nc_proxy.c

```diff
@@ -134,6 +134,9 @@ proxy_listen(struct context *ctx, struct conn *p)
         return NC_ERROR;
     }
 
+    if (pool->reuseport)
+        nc_set_reuseport(p->sd);
```

Comment:
1. This conditional can be moved into `proxy_reuse` - if reuseport config is set and this fails, then NC_ERROR should be returned
2. This only seems to have a documented effect for AF_INET/AF_INET6 https://man7.org/linux/man-pages/man7/socket.7.html

```
    switch (p->family) {
    case AF_INET:
    case AF_INET6:
        status = nc_set_reuseaddr(p->sd);
        break;
```

Reply:
I moved to be the same logic as for some other socket options, like `nc_set_nonblocking` or so.